iceoryx2-cli/lib/src/config_descriptions.rs

```diff
@@ -0,0 +1,327 @@
+                },
+                CliEntry {
+                    key: "defaults.publish-subscribe.publisher-max-loaned-samples",
+                    value_type: "[int]",
+                    description: "Maximum samples a publisher can loan.",
+                },
+                CliEntry {
+                    key: "defaults.publish-subscribe.publisher-history-size",
+                    value_type: "[int]",
+                    description: "Maximum history size a subscriber can request.",
+                },
+                CliEntry {
+                    key: "defaults.publish-subscribe.enable-safe-overflow",
+                    value_type: "[`true`|`false`]",
+                    description: "Default overflow behavior.",
+                },
+                CliEntry {
+                    key: "defaults.publish-subscribe.unable-to-deliver-strategy",
+                    value_type: "[`Block`|`DiscardSample`]",
+                    description: "Default strategy for non-overflowing setups when delivery fails.",
+                },
+                CliEntry {
+                    key: "defaults.publish-subscribe.subscriber-expired-connection-buffer",
+                    value_type: "[int]",
+                    description: "Expired connection buffer size of the subscriber. Connections to publishers are expired when the publisher disconnected from the service and the connection contains unconsumed samples.",
+                },
+            ],
+        },
+        CliSection {
+            name: "Defaults: Event Messaging Pattern",
+            entries: vec![
+                CliEntry {
+                    key: "defaults.event.max-listeners",
+                    value_type: "[int]",
+                    description: "Maximum number of listeners.",
+                },
+                CliEntry {
+                    key: "defaults.event.max-notifiers",
+                    value_type: "[int]",
+                    description: "Maximum number of notifiers.",
+                },
+                CliEntry {
+                    key: "defaults.event.max-nodes",
+                    value_type: "[int]",
+                    description: "Maximum number of nodes.",
+                },
+                CliEntry {
+                    key: "defaults.event.event-id-max-value",
+                    value_type: "[int]",
+                    description: "Greatest value an [`EventId`] can have.",
+                },
+                CliEntry {
+                    key: "defaults.event.deadline",
+                    value_type: "[int]",
+                    description: "Maximum allowed time between two consecutive notifications. If not sent after this time, all listeners attached to a WaitSet will be notified.",
+                },
+                CliEntry {
+                    key: "defaults.event.notifier-created-event",
+                    value_type: "[Option<int>]",
```

Comment:
Food for thought. As a user unfamiliar with the project I would be wondering how the value would look like. Just an `123` or maybe `Some(123)` or something else?

What do you think about adding an additional field like
```rs
key: ...
required: false // alternatively: 'optional: true'
value_type: "int"
```